clients/v1/clients.proto

```diff
@@ -289,3 +291,104 @@ message RoleResource {
   // A human readable name for the resource
   string display_name = 2;
 }
+
+service ServiceAccessTokensService {
+  // CreateServiceAccessToken creates a new service access token.
+  // A client can only create service access tokens for services granted via scopes,
+  // e.g. "sams::service_access_token.analytics::write" allows creating service access
+  // tokens for the Sourcegraph Analytics service. Service access token can only have scopes that
+  // belong to the same service, e.g. "analytics::analytics::read" when the service is
+  // "analytics".
+  rpc CreateServiceAccessToken(CreateServiceAccessTokenRequest) returns (CreateServiceAccessTokenResponse) {
+    option (sams_required_scopes) = "sams::service_access_token::write";
```

Comment:
_Catching up haven't looked at the implementation PRs:_

Do we intend to enforce granular scope (as said in the docstring , e.g. `"sams::service_access_token.analytics::write"`) or a global `sams::service_access_token::write` (as used with `sams_required_scopes`)? If we do intend to enforce granular scope, we can't really take advantage/use of `sams_required_scopes` because according to [SAMS token scope specification](https://www.notion.so/sourcegraph/SAMS-token-scope-specification-9c4ad28f4e394383ba697be7b02bb1c4), when a client grants `sams::service_access_token::write`, any granular scope is meaningless (because inherited).

If we intend to enforce with a single global scope, then we need to update the docstring here and other places to reflect that.